### DISCUSSION TITLE
Preferring most specific default value

### DISCUSSION THREAD

**Discussion by andrewzhurov** [cue-lang/cue#1540] [Ideas] [answered]
> CUE also bails out and requires explicit values if two conflicting defaults are specified for the same field, again limiting the search space.
> [source](https://cuelang.org/docs/concepts/logic/#relation-to-inheritance)

Would preferring the most specific default value be a fine resolution strategy for such a conflict?

**Comment by mpvl** [comment, marked as answer]
In the current spec and implementation, only "top-level" defaults are considered.
So for
```
bar: foo | *1
foo: int | *2
```
`bar` will resolve to `1`. In other words, the default mark in `bar` "erases" any of the default marks used in those referenced by disjuncts.

If a disjunction does not specify any default, the default of the reference is still considered.  So for
```
bar: foo | 1
foo: int | *2
```
`bar` will resolve to `2`, so allowing `foo` to carry over the default.

This gets some of the behavior you are after, I _think_, while actually further reducing the "search space" for default values even further than originally foreseen. This change was made a little while back and solves various other issues with the older semantics.
